arch/x86/core/ia32/gdbstub.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_internal.h>
+#include <ia32/exception.h>
+#include <inttypes.h>
+#include <debug/gdbstub.h>
+
+
+extern int z_gdb_main_loop(struct gdb_ctx *ctx, bool start);
```

Comment:
this must be in a header somewhere. probably some private header space for the kernel part of the GDB stub.

Reply:
will make it private

Reply:
I see more this pattern than having private headers. At least in this region. Where exactly you propose put this private header ? Together with the architecture specific code ? I'm sending a new version without change it because it is not clear to me that is the best thing and I want to address the other items + some extra things.

Reply:
Maybe add this into `include/debug/gdbstub.h`? Other architectures (hopefully) will have GDB stub in the future so it might be a good idea to define some common function prototypes as a starting point.

Reply:
I suggest `kernel/include/kernel_internal.h` since this API only gets invoked from the `arch/` code and not from the application. I don't care too much where it goes, but we're not going to have random extern declarations in C code